libs/@local/harpc/client/typescript/src/types/ResponseKind.ts

```diff
@@ -126,31 +127,28 @@ export type ResponseKind = Ok | Err;
 
 export type EncodeError = Effect.Effect.Error<ReturnType<typeof encode>>;
 
-export const encode = encodeDual(
-  (buffer: Buffer.WriteBuffer, kind: ResponseKind) =>
-    Effect.gen(function* () {
-      // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      if (isOk(kind)) {
-        return yield* Buffer.putU16(buffer, 0);
-      }
-
-      return yield* ErrorCode.encode(buffer, kind.code);
-    }),
+export const encode = implEncode((buffer, kind: ResponseKind) =>
+  // eslint-disable-next-line @typescript-eslint/no-use-before-define
+  isOk(kind)
```

Comment:
What's the motivation for having the eslint disable here (and above on `OkProto`) rather than moving `isResponseKind`, `isOk`, `isErr` above `OkProto`? I appreciate we do this in other files also, just curious, it isn't causing any practical issue.

Reply:
I have the files structured into sections, putting the function at the top was confusing to me when I did it first, it's always:

```
TypeId

Interface

Proto

Functions
	make
	(encode if available)
	(decode if available)
	... other functions
	isX function
	arbitrary
```

I have yet to enforce that order - I don't know how to^^ (but most files should follow it tho)